platform/viewer/src/studylist/StudyListRoute.js

```diff
@@ -78,55 +131,54 @@ function StudyListRoute(props) {
     setActiveModalId('DicomStorePicker');
   }
 
-  // Called when relevant state/props are updated
-  // Watches filters and sort, debounced
-  useEffect(
-    () => {
-      const fetchStudies = async () => {
-        try {
-          setSearchStatus({ error: null, isSearchingForStudies: true });
-
-          const response = await getStudyList(
-            server,
-            debouncedFilters,
-            debouncedSort,
-            rowsPerPage,
-            pageNumber,
-            displaySize
-          );
-
-          setStudies(response);
-          setSearchStatus({ error: null, isSearchingForStudies: false });
-        } catch (error) {
-          console.warn(error);
-          setSearchStatus({ error: true, isFetching: false });
-        }
-      };
+  const fetchData = useCallback(
+    debounce(async newStudyListSessionProps => {
+      setSearchStatus({ error: null, isSearchingForStudies: true });
+      const {
+        sort,
+        filters,
+        rowsPerPage,
+        pageNumber,
+      } = newStudyListSessionProps;
+
+      const response = await getStudyList(
+        server,
+        filters,
+        sort,
+        rowsPerPage,
+        pageNumber,
+        displaySize
+      );
+
+      setStudies(response);
+      setSearchStatus({ error: null, isSearchingForStudies: false });
+    }, FETCH_DEBOUNCE_TIME),
+    [server, displaySize]
+  );
 
-      if (server) {
-        fetchStudies();
+  const fetchStudies = useCallback(
+    async (newStudyListSessionProps, forceFlush = false) => {
+      if (!server) {
+        return;
+      }
+
+      try {
+        setStudyListSessionProps(newStudyListSessionProps);
+        fetchData(newStudyListSessionProps);
+        if (forceFlush) {
+          fetchData.flush();
+        }
+      } catch (error) {
+        console.warn(error);
+        setSearchStatus({ error: true, isFetching: false });
       }
     },
-    // TODO: Can we update studies directly?
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [
-      debouncedFilters,
-      debouncedSort,
-      rowsPerPage,
-      pageNumber,
-      displaySize,
-      server,
-    ]
+    [server, displaySize]
```

Comment:
Does `useCallback` net us much of a benefit here? Is the readability loss worth the performance gain?

Reply:
useCallback brings us mainly two things:
- performance
- unicity

From my understanding, considering React`s new way of coding, this method is not less readable, is it? We can change, add comments :D, your call my friend